R/import-standalone-obj-type.R

```diff
@@ -0,0 +1,348 @@
+# Standalone file: do not edit by hand
```

Comment:
One of your other PRs is pulling in `rlang`; does this file go away with an `rlang` dependency?

Reply:
We haven't yet standardised this interface, so we're currently vendoring it into every package where we use it. I think it's worth the cost, because it makes input validation _so_ much more pleasant. At some point once we provide an exported interface, we'll be able to remove from this and other packages.